### PR TITLE
PCC58 yashaslokesh

### PR DESCRIPTION
**Pull request by yashaslokesh**
ATTENTION: before clicking "Create Pull Request" please submit some meta data, thanks!

Difficulty level (1-10): [5]
Estimated time spent (hours): [4]
Completed (yes/no): [yes]
I stretched my coding skills (if yes what did you learn?): [Used itertools.chain because some Talk Python transcripts were missing, so I had to skip those. I also used sqlite3 for the first time. It's really easy to use, I think I'll use it more in the future to store data that I retrieve from APIs. I also got more experience using the GitHub API for retrieving repo content.]
Other feedback (what can we improve?): []

### CODE REVIEW

File: 58/yashaslokesh/transcripts.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+"""
+    Generator function that will yield the transcript number and transcript for every
+    available transcript. This is done so that the strings don't take up too much memory
+    all at once and slow down the script (subsequent operation of adding text to sqlite)
+"""
+def transcripts(url):
+    for num in it.chain(range(0, FIRST_SET_TOTAL_TRANSCRIPTS + 1), 
+                        range(SECOND_SET_START, SECOND_SET_END + 1)):
+        # Creates file names going 000.txt, ...014.txt, ... 186.txt
+        transcript_num = f'{str(num).zfill(3)}.txt'
```
Comment (42B) [inline 58/yashaslokesh/transcripts.py]:
```python
>>> num = 1
>>> f'{num:0>3}'
'001'
```

File: 58/yashaslokesh/transcripts.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        headers = {'Authorization': f'token {os.environ.get("TPT_OAUTH_TOKEN")}'}
+        result = requests.get(url + transcript_num, headers=headers)
+
+        # Obtain the textual portion of the file
+        data = json.loads(result.text)['content']
```
Comment (42B) [inline 58/yashaslokesh/transcripts.py]:
```python
data = result.json()['content']
```

File: 58/yashaslokesh/transcripts.py (hunk 1 above)
Comment (yashaslokesh) [reply 58/yashaslokesh/transcripts.py]: I didn't know you could do this, is this mentioned somewhere in the python docs?

File: 58/yashaslokesh/transcripts.py (hunk 2 above)
Comment (yashaslokesh) [reply 58/yashaslokesh/transcripts.py]: That's right, I forgot a response object has the .json() method. Thanks

File: 58/yashaslokesh/transcripts.py (hunk 1 above)
Comment (42B) [reply 58/yashaslokesh/transcripts.py]: https://docs.python.org/3/library/string.html#format-specification-mini-language

File: 58/yashaslokesh/transcripts.py (hunk 2 above)
Comment (42B) [reply 58/yashaslokesh/transcripts.py]: @yashaslokesh 

No problem!

Comment (pybites) [reply 58/yashaslokesh/transcripts.py]: Nice thing about `requests` indeed

File: 58/yashaslokesh/transcripts.py (hunk 1 above)
Comment (pybites) [reply 58/yashaslokesh/transcripts.py]: @42B thanks for reviewing, how did you land on this PR? Just curious

**Comment by pybites** [comment]
Great work (will review code at a later point because I still need to code this one!)
Keep the PRs coming :)

**Pull request merged**